lib/core/service/lantern_ffi_service.dart

```diff
@@ -61,6 +68,57 @@ class LanternFFI {
     //SystemChannels.platform.invokeMethod('SystemNavigator.pop');
   }
 
+  // Initialize the system proxy isolate
+  static Future<void> _initializeSystemProxyIsolate() async {
+    final receivePort = ReceivePort();
+    // create isolate that listens for system proxy commands
+    await Isolate.spawn(_proxyIsolateEntry, receivePort.sendPort);
+    _proxySendPort = await receivePort.first;
+    _isolateInitialized.complete();
+  }
+
+  // initialize the isolate if need be and send the vpnStatus to it
+  static Future<String> sendVpnStatus(String vpnStatus) async {
+    if (!_isolateInitialized.isCompleted) {
+      await _initializeSystemProxyIsolate();
+    }
+
+    final responsePort = ReceivePort(); // Port to receive isolate's response
+    _proxySendPort?.send([vpnStatus, responsePort.sendPort]);
+
+    // Listen for the result (success or error)
+    final message = await responsePort.first;
+    if (message == "done") {
```

Comment:
Before we are not checking error and done from isolate now we are doing it.

Reply:
Just to give some context: I thought we could avoid some redundant response checking by sending the status update (if setting the system proxy succeeded or failed) via the websocket channel afterwards (actually, I don't think we need to do that anymore and can simplify things further with this setup).

Using a ReceivePort here to check if the isolate _received/processed_ the request though definitely makes sense. Thanks for making the change!